src/OrchardCore.Modules/OrchardCore.GitHub/Extensions/OrchardCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using Microsoft.Extensions.Configuration;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.GitHub.Settings;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class OrchardCoreBuilderExtensions
+    {
+        public static OrchardCoreBuilder ConfigureGitHubSettings(this OrchardCoreBuilder builder)
+        {
+            builder.ConfigureServices((tenantServices, serviceProvider) =>
+            {
+                var configurationSection = serviceProvider.GetRequiredService<IShellConfiguration>().GetSection("OrchardCore_GitHub");
+
+                tenantServices.PostConfigure<GitHubAuthenticationSettings>(settings => configurationSection.Bind(settings));
```

Comment:
Can you set `GitHubAuthenticationSettings.CallbackPath` from JSON? Same for `GoogleAuthenticationSettings`.

Reply:
You can pass `CallbackPath` from both admin area settings and `appsettings.json`. Did you mean something else?

Reply:
I need to check, also I might check reverse proxy, because it requires an enum